File: backend/workspaces/01473b7e-0231-40f5-a14b-2433a1ca88d1/backend/task.py
# app/models/task.py
from datetime import datetime
from enum import Enum
from typing import Optional

class TaskStatus(str, Enum):
    pending = "pending"
    completed = "completed"

class Task:
    def __init__(
        self,
        id: int,
        title: str,
        description: Optional[str] = None,
        status: TaskStatus = TaskStatus.pending,
        created_at: datetime = datetime.now(),
        updated_at: datetime = datetime.now(),
    ):
        self.id = id
        self.title = title
        self.description = description
        self.status = status
        self.created_at = created_at
        self.updated_at = updated_at

    def update(self, **kwargs):
        for key, value in kwargs.items():
            if hasattr(self, key):
                setattr(self, key, value)
        self.updated_at = datetime.now()
        return self

File: backend/workspaces/2935fdcd-3520-441c-86bb-9f864f82290e/backend/task.ts
// TypeScript type definitions for tasks
export type TaskStatus = 'pending' | 'completed';

export interface Task {
  id: number;
  title: string;
  description?: string;
  status: TaskStatus;
}

export interface TaskCreate {
  title: string;
  description?: string;
}

export interface TaskUpdate {
  title?: string;
  description?: string;
  status?: TaskStatus;
}